lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2544,6 +2659,25 @@ namespace {
 
         auto VD = cast<VarDecl>(member);
 
+        if (isa<clang::IndirectFieldDecl>(nd) || decl->isUnion()) {
+          // Don't import unconditionnaly unavailable fields that have
+          // no associated storage.
+          bool isUnconditionallyUnavailable = false;
+          auto attrs = VD->getAttrs().getAttributes<AvailableAttr, false>();
+
+          for (auto attr : attrs) {
+            if (attr->isUnconditionallyUnavailable()) {
+              isUnconditionallyUnavailable = true;
+              break;
+            }
+          }
+          if (isUnconditionallyUnavailable) {
+            continue;
+          }
+        }
```

Comment:
I actually repushed with `isUnavailable()` since after a few manual tests it does not seem to affect error reporting (I was worried that the error message for accessing an unimported field would be different from the one we get for an unavailable field, but it does not seem to be the case).